samples/nrf9160/nrf_cloud_rest_device_message_tol2/README.rst

```diff
@@ -0,0 +1,87 @@
+.. _nrf_cloud_rest_device_message_replicate:
+
+nRF9160: nRF Cloud REST Device Message
+######################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The REST Device Message sample demonstrates how to use the `nRF Cloud REST API`_ to send `Device Messages <nRF Cloud Device Messages_>`_ using the ``SendDeviceMessage`` REST endpoint.
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-sample-yaml::
+
+.. include:: /includes/tfm.txt
+
+The sample requires an `nRF Cloud`_ account.
+
+Your device must be provisioned with `nRF Cloud`_. If it is not, follow the instructions in `Provisioning <nrf_cloud_rest_device_message_replicate_provisioning_>`_.
+
+.. note::
+   This sample requires modem firmware v1.3.x or later.
+
+Limitations
+***********
+
+The `nRF Cloud REST API`_ requires all requests to be authenticated with a JSON Web Token (JWT).
+See  `nRF Cloud Security`_ for more details.
+Generating valid JWTs requires the network carrier to provide date and time to the modem, so the sample must first connect to an LTE carrier and determine the current date and time before REST requests can be sent.
+
+Note also that the `nRF Cloud REST API`_ is stateless.
+This differs from the `nRF Cloud MQTT API`_, which requires you to establish and maintain an `MQTT`_ connection while sending `Device Messages <nRF Cloud Device Messages_>`.
+
+User interface
+**************
+
+Once the device is provisioned and connected, it will periodically send messages to the cloud in 60 seconds.
```

Comment:
I would reword this to say "send messages to the cloud once every 60 seconds."

Reply:
Well, I meant to say that the service has a timeout of maximum 60 seconds, you cannot change it to send messages indefinitely. I'll change it to something more reasonable.

Reply:
Hmm.  OK, actually, your main() function has a loop which sends messages until there is an error, once every 3 seconds.

Reply:
Yes, the error will happens eventually, since I can set the timeout in rest-context but if it's higher than 60,000ms, it will be set to the value of 60,000 in runtime; also setting it 0 will make the timeout immediately.

Reply:
The 60 seconds is used in case the server does not respond to the REST call, but normally it will respond to every call within a few seconds.  I don't feel it is useful to say "in 60 seconds."

The "happy case" is that a message will be sent every 3 seconds forever.